src/dataset.jl

```diff
@@ -59,17 +59,40 @@ KeyedDataset(; constraints=Pattern[], kwargs...) = KeyedDataset(kwargs...; const
 
 function Base.show(io::IO, ds::KeyedDataset{K, T}) where {K, T}
     n = length(ds.data)
-    print(io, "KeyedDataset{$K, $T} with $n entries:")
-    for c in ds.constraints
-        print(io, "\n  ", c)
-    end
+    m = length(ds.constraints)
+
+    # Extract the constraints as a vector for indexing
+    constraints = collect(ds.constraints)
+
+    lines = String["KeyedDataset with:", "  $n components"]
     for (k, v) in ds.data
-        printstyled(io, "\n  ", k, " => "; color=:cyan)
-        printstyled(io, replace(sprint(Base.summary, v), "\n" => "\n    "); color=:cyan)
-        print(io, "\n    ", replace(sprint(Base.print_array, v), "\n" => "\n    "))
+        # Identify shared dimensions where appropriate
+        dimensions = map(dimnames(v)) do dimname
+            cidx = findall(c -> (k..., dimname) in c, constraints)
+            isempty(cidx) ? string(dimname) : string(dimname, "[", _only(cidx), "]")
```

Comment:
What happens if a dimension satisfies more than 1 constraint?

Reply:
Yes, Glenn mentioned that in #29 as well.

> Currently, yes, but my plan was to implement some sort of Pattern reduction algorithm because logically if axis x must align with all axes in pattern a and b then that must mean that all axes in a and b must align with each other and can be described by a single Pattern

Reply:
Made an issue https://github.com/invenia/AxisSets.jl/issues/36